conans/client/conan_api.py

```diff
@@ -842,6 +842,37 @@ def export(self, path, name, version, user, channel, keep_source=False, cwd=None
             lockfile_out = _make_abs_path(lockfile_out, cwd)
             graph_lock_file.save(lockfile_out)
 
+    @api_method
+    def export_all(self, path, name, version, user, channel, keep_source=False, cwd=None,
+                   lockfile=None, lockfile_out=None, ignore_dirty=False):
+        graph_lock, graph_lock_file = None, None
+        if lockfile:
+            lockfile = _make_abs_path(lockfile, cwd)
+            graph_lock_file = GraphLockFile.load(lockfile, self.app.config.revisions_enabled)
+            graph_lock = graph_lock_file.graph_lock
+            self.app.out.info("Using lockfile: '{}'".format(lockfile))
+
+        self.app.load_remotes()
+
+        from concurrent.futures import as_completed, ThreadPoolExecutor
+        from pathlib import Path
+
+        futures = []
+        path = Path(path)
```

Comment:
We probably want to keep the code Py2 compatible, it won't cost much to use old style ``os.path`` syntax to work.